modules/users/tests/server/user.server.routes.tests.js

```diff
@@ -262,6 +262,601 @@ describe('User CRUD tests', function () {
+            currentPassword: 'some_wrong_passwordAa$'
+          })
+          .expect(400)
+          .end(function (err, res) {
+            if (err) {
+              return done(err);
+            }
+
+            res.body.message.should.equal('Current password is incorrect');
+            return done();
+          });
+      });
+  });
+
+  it('should not be able to change user own password if no new password is at all given', function (done) {
+    agent.post('/api/auth/signin')
+      .send(credentials)
+      .expect(200)
+      .end(function (signinErr, signinRes) {
+        // Handle signin error
+        if (signinErr) {
+          return done(signinErr);
+        }
+
+        // Change password
+        agent.post('/api/users/password')
+          .send({
+            newPassword: '',
+            verifyPassword: '',
+            currentPassword: credentials.password
+          })
+          .expect(400)
+          .end(function (err, res) {
+            if (err) {
+              return done(err);
+            }
+
+            res.body.message.should.equal('Please provide a new password');
+            return done();
+          });
+      });
+  });
+
+  it('should not be able to change user own password if no new password is at all given', function (done) {
+
+    // Change password
+    agent.post('/api/users/password')
+      .send({
+        newPassword: '1234567890Aa$',
+        verifyPassword: '1234567890Aa$',
+        currentPassword: credentials.password
+      })
+      .expect(400)
+      .end(function (err, res) {
+        if (err) {
+          return done(err);
+        }
+
+        res.body.message.should.equal('User is not signed in');
+        return done();
+      });
+  });
+
+  it('should be able to get own user details successfully', function (done) {
+    agent.post('/api/auth/signin')
+      .send(credentials)
+      .expect(200)
+      .end(function (signinErr, signinRes) {
+        // Handle signin error
+        if (signinErr) {
+          return done(signinErr);
+        }
+
+        // Get own user details
+        agent.get('/api/users/me')
+          .expect(200)
+          .end(function (err, res) {
+            if (err) {
+              return done(err);
+            }
+
+            res.body.should.be.instanceof(Object);
+            res.body.username.should.equal(user.username);
+            res.body.email.should.equal(user.email);
+            should.not.exist(res.body.salt);
+            should.not.exist(res.body.password);
+            return done();
+          });
+      });
+  });
+
+  it('should not be able to get any user details if not logged in', function (done) {
+    // Get own user details
+    agent.get('/api/users/me')
+      .expect(200)
+      .end(function (err, res) {
+        if (err) {
+          return done(err);
+        }
+
+        should.not.exist(res.body);
+        return done();
+      });
+  });
+
+  it('should be able to update own user details', function (done) {
```

Comment:
the `user.save()` part don't need to apply to all tests so we can't just throw that in `beforeEach`, and there's no sense to wrap it with any other function as it's anyway just going to get a callback.

I was thinking maybe wrapping the user signing-in in a function but it looks more readable and extensible as is.